test.js

```diff
@@ -288,6 +288,15 @@ test('execa() returns a promise with kill() and pid', t => {
 	t.is(typeof pid, 'number');
 });
 
+test('child_process.spawn() propagated errors have correct shape', t => {
+	const cp = execa('noop', {uid: -1});
+	t.notThrows(() => {
+		cp.catch(() => {});
+		cp.unref();
+		cp.on('error', () => {});
+	});
```

Comment:
This test call methods from `Promise`, `ChildProcess` and `EventEmitter`. Just checking they exist is not enough. For example, `this` could be incorrect, making those methods crash even if they exist.